protocols/discovery/src/substream.rs

```diff
@@ -378,7 +378,7 @@ impl Substream {
                 .iter()
                 .map(|address| multiaddr_to_socketaddr(address).unwrap())
                 .filter_map(|address| {
-                    if !address.ip().is_loopback() || RawAddr::from(address).is_reachable() {
+                    if RawAddr::from(address).is_reachable() || !address.ip().is_loopback() {
```

Comment:
Why exchange the order? Does `is_reachable` only analyze the address or do real dialing to check whether it is reachable? If the answer is the latter, I believe `is_loopback` is cheaper and should come first.

Reply:
`is_reachable` only analyze the address.

Here we need to reject the unconnectable address and keep the 0.0.0.0 address.
This condition is still a bit problem, it would be better to directly indicate the special address.

Reply:
In the discussion in https://github.com/nervosnetwork/ckb/pull/1413#discussion_r314934305 , it seems the agreement is changing to `RawAddr::from(address).is_reachable() && !address.ip().is_loopback()`

BTW, this should not be messed up in a version bump PR.

Reply:
No, `RawAddr::from(address).is_reachable() && !address.ip().is_loopback()` is wrong

I don't want to argue in comment, but this condition is wrong.

That pr https://github.com/nervosnetwork/ckb/pull/1413 will not be merged, unless I agree, it changes the behavior of the test.

Reply:
make this change to another pr